src/sentry/api/endpoints/auth_index.py

```diff
@@ -115,7 +115,12 @@ def put(self, request):
 
         # attempt password authentication
         else:
-            authenticated = request.user.check_password(validator.validated_data["password"])
+            authenticated = (
+                authenticate(
+                    username=request.user.username, password=validator.validated_data["password"]
+                )
+                is not None
```

Comment:
@BYK if the `authenticate()` function will return `False`, value of `authenticated` will be `True`. I don't think it is the right behavior here.

Reply:
What I'm told by @mattrobenolt was `authenticate` returning a `User` model or `None` so this should be okay. The docs also show a very similar use here: https://docs.djangoproject.com/en/1.11/topics/auth/default/#authenticating-users

Reply:
Oh, I was thinking that this function can return `False`, but you're right.